src/wkok/openai_clojure/azure/api.clj

```diff
@@ -0,0 +1,38 @@
+(ns wkok.openai-clojure.azure.api
+  (:require [wkok.openai-clojure.azure.core :as core]))
+
+
+(defn create-completion
```

Comment:
Unfortunate naming by Azure, but I think we should stick as close to their spec as possible so maybe `completions-create`  for the function name

Reply:
I think one goal of this library could be that tool programmers would use it, so their tools would work with Azure OpenAI and openai in the same way.
( I assume hat Azure will stick to the openai API semantics, and only have the small details different we see so far, but this is indeed not guaranteed)

Reply:
The responses are "identical":
So I think this goal is possible to reach.
But then the 2 namespaces approach is not the best way.

;; Azure openai
;; => {:id "cmpl-6jxrzB6in83RE2XRANbCWpq3GIX8C",
;;     :object "text_completion",
;;     :created 1676412919,
;;     :model "text-davinci-003",
;;     :choices
;;     [{:text "\n\nI'm good, thank you. How about you?",
;;       :index 0,
;;       :logprobs nil,
;;       :finish_reason "stop"}],
;;     :usage {:prompt_tokens 4, :completion_tokens 13, :total_tokens 17}})


;; openai
;; {
;;   "id": "cmpl-uqkvlQyYK7bGYrRHQ0eXlWi7",
;;   "object": "text_completion",
;;   "created": 1589478378,
;;   "model": "text-davinci-003",
;;   "choices": [
;;               {
;;                 "text": "\n\nThis is indeed a test",
;;                 "index": 0,
;;                 "logprobs": null,
;;                 "finish_reason": "length"}]
    
;;   ,
;;   "usage": {
;;             "prompt_tokens": 5,
;;             "completion_tokens": 7,
;;             "total_tokens": 12}}

Reply:
OK makes sense then to keep the fn names matching those of OpenAI

I think the 2 namespace approach still have benefits

- it's a minor `(require)` change for the developer who wants to switch between azure & openai
- while still achieving a smooth initialization, no need for calling special init functions
- keeps the code cleaner with no awkward conditional branches everywhere depending on which service is used

Reply:
I am "expecting" that a some  people will develop graphic tools based on the openai API,
and they might want to support both current "implementations" of the same API, which differ so far only in teh "authentication" and the "endpoint" setup. I believe that Microsoft will continue doing this as well in teh future.
Maybe even other "providers" of open compatibel APIs will appear.

My proposal would is to have a single api namespace (as current).

and add for every function +1 arrity, where a keyword is teh selector of the implementation (and it defaults to openai),

to be called like this:

```
(create-completion :azure-openai params)
(create-completion :openai params)
(create-completion params) (defaults to  :openai)
```

The initialization could happen "lazily" with a `(delay )` , to avoid the need of having all the environment variables present all the time.
Probably even a "martian initialization on each function call" would be fine compared to the slowness of the API call, which takes seconds. But I have not timed it.

Reply:
Yes I think that could work.

- Single `api` namespace (maybe even drop the `core` namespace completely)
- Multi arity operation functions defaulting the implementation to `:openai`

For initialization, we could define 2 atoms `openai-martian` and `azure-martian` and have each operation function initialise the martian only if it's not already initialised, by checking / resetting the atom accordingly

Reply:
Ok, I will try it.
Just wondering about "one atom" vs  "2 delays"..
I will try both and see.

Reply:
Think either atom / delay approach would be fine, whichever results in the simplest code